### PR TITLE
docs: add cli config reference

### PR DESCRIPTION
**Pull request by lornajane**
## What/Why/How?

We have some config options that are in use in the CLI, but the rest of the config docs are still private for the new products. Create matching-format pages for the options that belong here so we can link to them.

## Check yourself

- [x] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests
- [x] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: docs/configuration/reference/decorators.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+The example includes adding the plugin, partly to remind you that this is also needed.
+
+## Related options
+
+- [apis](./apis.md) configuration options allow setting per-API configuration in `redocly.yaml`.
+- [rules](./rules.md) settings define the linting rules that are used.
+
+## Resources
+
+- Learn more about [decorators](../../decorators.md).
```
Comment (tatomyr) [inline docs/configuration/reference/decorators.md:89]: Do we maybe need to add a link to the plugins detailed explanation here?

File: docs/configuration/index.md [hunk 2]
```diff
@@ -213,7 +213,7 @@ Redocly CLI supports one `http` header per URL.
 
 Here is an example for adding header definitions:
 
-```yaml
+```text
```
Comment (tatomyr) [inline docs/configuration/index.md:216]: What's the difference? The example below is a proper `yaml` as far as I can judge.

Comment (lornajane) [reply docs/configuration/index.md:216]: The syntax highlighter thinks that the `/*` bits are a comment and turns half of it grey! Our yaml highlight is so plain that the text isn't much different so I flipped it.

File: docs/configuration/reference/apis.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+- rules
+- [Rules object](./rules.md)
+- Additional rule configuration for this API.
+
+---
+
+- decorators
+- [Decorators object](./decorators.md)
+- Additional decorator configuration for this API.
+
```
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/configuration/reference/apis.md:48]: I'll add it.

File: docs/configuration/reference/decorators.md (hunk 1 above)
Comment (lornajane) [reply docs/configuration/reference/decorators.md:89]: Link added.

File: docs/configuration/reference/extends.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+- the name of a [built-in ruleset](../../rules.md#rulesets)
+- configuration defined in a [custom plugin](../../custom-plugins/index.md)
+- a path or URL to another `redocly.yaml` file
+
+Extends is useful if you use a common ruleset across multiple projects.
+Define a ruleset in one location, and each project can `extend` it, with or without modification.
+
+{% admonition type="info" name="Default ruleset: recommended" %}
+If `extends` isn't set, the [recommended ruleset](../../rules/recommended.md) is used by default.
```
[the remark this answers is not in the document]
Comment (lornajane) [reply docs/configuration/reference/extends.md:16]: Good catch! Fixed it.

File: docs/configuration/reference/apis.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+- Option
+- Type
+- Description
+
+---
+
+- `{name}@{version}`
+- [API object](#api-object)
+- Required. Each API needs a name and optionally a version. Supports alphanumeric characters and underscores.
```
Comment (HCloward) [inline docs/configuration/reference/apis.md:19]: Should we make the Required all caps and bolded as in the other pages in the beta docs?

File: docs/configuration/reference/apis.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+- decorators
+- [Decorators object](./decorators.md)
+- Additional decorator configuration for this API.
+
+---
+
+- preprocessors
+- [Decorators object](./decorators.md)
+- Preprocessors run before linting, and follow the same structure as decorators. We recommend use of decorators in most cases.
```
Comment (HCloward) [inline docs/configuration/reference/apis.md:53]:
```suggestion
- Preprocessors run before linting, and follow the same structure as decorators. We recommend the use of decorators over preprocessors in most cases.
```

File: docs/configuration/reference/decorators.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+- {additional properties}
+- any
+- Some decorators support additional configuration, check the documentation for each specific decorator to discover the values that can be used. For example the [`filter-out` decorator](../../decorators/filter-out.md) supports `property` and `value` settings for what to filter.
+
+---
+
+- severity
+- string
+- Severity level of any problems reported with this decorator. Setting this option is usually only useful for troubleshooting purposes. The value must be one of `error`, `warn`, or `off`, where `off` disables the decorator.
```
Comment (HCloward) [inline docs/configuration/reference/decorators.md:43]: Is there a default?

Comment (lornajane) [reply docs/configuration/reference/decorators.md:43]: I'm not sure, I didn't know this existed until I tried to write the doc and found it!

File: docs/configuration/reference/apis.md (hunk 5 above)
Comment (lornajane) [reply docs/configuration/reference/apis.md]: Updated this everywhere.

File: docs/configuration/reference/plugins.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+A basic example of including two plugins from a directory named `plugins/` is shown in the example below:
+
+```yaml
+plugins:
+  - plugins/my-best-plugin.js
+  - plugins/another-plugin.js
+```
+
+Remember that you need to include plugins in the `plugins` section before you can use the contest of the plugin elsewhere in the configuration file.
```
Comment (HCloward) [inline docs/configuration/reference/plugins.md:23]: I don't think you mean contest here - maybe context?

File: docs/configuration/reference/extends.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+## Related options
+
+- [apis](./apis.md) configuration options allow setting per-API configuration in `redocly.yaml`.
+- [rules](./rules.md) settings define the linting rules that are used.
+
+## Resources
+
+- Detailed documentation and examples on [extending configuration](../extends.md)
```
Comment (HCloward) [inline docs/configuration/reference/extends.md:41]:
```suggestion
- Detailed documentation and examples on [extending configuration](../extends.md).
```

File: docs/configuration/reference/plugins.md (hunk 8 above)
Comment (lornajane) [reply docs/configuration/reference/plugins.md]: "content" - good spot!

**Review by HCloward — APPROVED** [review]
Everything looks good to me. I checked all links in sample preview and all work as expected.

**Pull request merged**